bundle/edu.gemini.itc/src/main/java/edu/gemini/itc/gmos/Gmos.java

```diff
@@ -18,79 +14,136 @@
-    protected int _spatialBinning;
-
-    protected boolean _IFUUsed = false;
-    protected String _IFUMethod;
-    protected double _IFUOffset;
-    protected double _IFUMinOffset;
-    protected double _IFUMaxOffset;
-    protected IFUComponent _IFU;
-    protected boolean _IFU_IsSingle = false;
 
     // These are the limits of observable wavelength with this configuration.
-    protected double _observingStart;
-    protected double _observingEnd;
 
     private int _detectorCcdIndex = 0; // 0, 1, or 2 when there are multiple CCDs in the detector
 
-    public Gmos(String FILENAME, String INSTUMENT_PREFIX, int detectorCcdIndex) throws Exception {
+    public Gmos(GmosParameters gp, ObservationDetailsParameters odp, String FILENAME, int detectorCcdIndex) throws Exception {
         super(INSTR_DIR, FILENAME);
-        _detectorCcdIndex = detectorCcdIndex;
 
-        // The instrument data file gives a start/end wavelength for
-        // the instrument.  But with a filter in place, the filter
-        // transmits wavelengths that are a subset of the original range.
+        this.odp    = odp;
+        this.gp     = gp;
+
+        _detectorCcdIndex = detectorCcdIndex;
 
-        _observingStart = super.getStart();
-        _observingEnd = super.getEnd();
         _sampling = super.getSampling();
 
-        INSTR_PREFIX = INSTUMENT_PREFIX;
+        // TODO: filter is not yet defined, need to work with filter from gp, clean this up
+        if (!gp.getFilter().equals(GmosNorthType.FilterNorth.NONE) && !gp.getFilter().equals(GmosSouthType.FilterSouth.NONE)) {
+            _Filter = Filter.fromWLFile(getPrefix(), gp.getFilter().name(), getDirectory() + "/");
+            addFilter(_Filter);
+        }
+
+
+        FixedOptics _fixedOptics = new FixedOptics(getDirectory() + "/", getPrefix());
+        addComponent(_fixedOptics);
+
+
+        //Choose correct CCD QE curve
+        switch (gp.getCCDtype()) {
+            // E2V, site dependent
+            case E2V:
+                switch (gp.getSite()) {
+                    // E2V for GN: gmos_n_E2V4290DDmulti3.dat      => EEV DD array
+                    case GN:
+                        _detector = new Detector(getDirectory() + "/", getPrefix(), "E2V4290DDmulti3", "EEV DD array");
+                        _detector.setDetectorPixels(DETECTOR_PIXELS);
+                        if (detectorCcdIndex == 0) _instruments = new Gmos[]{this};
+                        break;
+                    // E2V for GS: gmos_n_cdd_red.dat              => EEV legacy
+                    case GS:
+                        _detector = new Detector(getDirectory() + "/", getPrefix(), "ccd_red", "EEV legacy array");
+                        _detector.setDetectorPixels(DETECTOR_PIXELS);
+                        if (detectorCcdIndex == 0) _instruments = new Gmos[]{this};
+                        break;
+                    default:
+                        throw new IllegalArgumentException();
```

Comment:
Hm, `IllegalArgumentException` is intended to indicate an error made by the caller. I usually throw `Error` in cases like this, with an indication that the condition should be impossible; if you have exhausted all the members of the `enum` you're in pretty bad trouble.